lib/Graph/Graph.cpp

```diff
@@ -682,6 +682,35 @@ Node *Function::createChannelShuffle(llvm::StringRef name, NodeValue input,
   return createReshape(name.str() + ".reshape2", T, inDims);
 }
 
+Node *Function::createSqueeze(llvm::StringRef name, NodeValue input,
+                              llvm::ArrayRef<size_t> axes) {
+  auto originalSize = axes.size();
+  GLOW_ASSERT(originalSize > 0 && "Parameter `axes` must be provided.");
+
+  ShapeVector shapeAxes(axes.begin(), axes.end());
+  std::sort(shapeAxes.begin(), shapeAxes.end());
+  shapeAxes.erase(std::unique(shapeAxes.begin(), shapeAxes.end()),
+                  shapeAxes.end());
+  GLOW_ASSERT(shapeAxes.front() >= 0 && "Dimension ids must be non-negative.");
+
+  auto inDims = input.dims();
+  GLOW_ASSERT(
+      shapeAxes.size() < inDims.size() &&
+      "The size of dimenstions to squeeze must less than the input size");
+  ShapeVector newDims;
+  int j = 0;
+  for (int i = 0; i < inDims.size(); i++) {
+    if (j < shapeAxes.size() && shapeAxes[j] == i) {
+      GLOW_ASSERT(inDims[i] == 1 && "The dimension to sequeeze must be 1.");
+      j++;
+    } else {
+      newDims.push_back(inDims[i]);
+    }
+  }
+  return createReshape(name.str() + ".reshape", input,
+                       (llvm::ArrayRef<size_t>)newDims);
```

Comment:
Is typecast to `(llvm::ArrayRef<size_t>)` necessary?

Reply:
no :P